pynest/nest/lib/hl_api_helper.py

```diff
@@ -528,7 +528,17 @@ def restructure_data(result, keys):
     int, list or dict
     """
     if is_literal(keys):
-        final_result = result[0] if len(result) == 1 else list(result)
+        if len(result) != 1:
+            all_keys = sorted({key for result_dict in result for key in result_dict})
```

Comment:
# :exploding_head:

Reply:
yeah, it also took me a good 30 s to figure out all that was happening in this single line...

Reply:
Do you want me to break it up into more steps?

Reply:
no no, it's fine ^^